centreon/src/CentreonUser/Domain/Repository/TimeperiodRepository.php

```diff
@@ -57,24 +88,30 @@ public function checkListOfIds(array $ids): bool
     public function getPaginationList($filters = null, int $limit = null, int $offset = null, $ordering = []): array
     {
         $sql = 'SELECT SQL_CALC_FOUND_ROWS `tp_id`, `tp_name`, `tp_alias` '
-            . 'FROM `' . $this->getClassMetadata()->getTableName() . '`';
+            . 'FROM `:db`.`timeperiod`';
 
         $collector = new StatementCollector();
 
         $isWhere = false;
         if ($filters !== null) {
-            if (array_key_exists('search', $filters) && $filters['search']) {
+            if (
+                array_key_exists('search', $filters)
+                && $filters['search']
+            ) {
                 $sql .= ' WHERE (`tp_name` LIKE :search OR `tp_alias` LIKE :search)';
                 $collector->addValue(':search', "%{$filters['search']}%");
                 $isWhere = true;
             }
 
-            if (array_key_exists('ids', $filters) && is_array($filters['ids'])) {
+            if (
+                array_key_exists('ids', $filters)
+                && is_array($filters['ids'])
+            ) {
```

Comment:
Missing test about empty array like other location of this pr ?
```suggestion
            if (
                array_key_exists('ids', $filters)
                && is_array($filters['ids'])
                && [] !== $filters['ids']
            ) {
```